xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -22,6 +22,76 @@ use sp_std::{borrow::Borrow, marker::PhantomData};
 use xcm::latest::prelude::*;
 use xcm_executor::traits::Convert;
 
+/// Prefix for bytes in account creation of foreign chains
+pub const FOREIGN_CHAIN_PREFIX: [u8; 30] = *b"ForeignChainAliasAccountPrefix";
+
+/// This converter will for a given `AccountId32`/`AccountKey20`
+/// always generate the same "remote" account for a specific
+/// sending chain.
+/// I.e. the user gets the same remote account
+/// on every consuming para-chain and relay chain.
+///
+/// Can be used as a converter in `SovereignSignedViaLocation`
+pub struct ForeignChainAliasAccount<AccountId>(PhantomData<AccountId>);
+impl<AccountId: From<[u8; 32]> + Clone> Convert<MultiLocation, AccountId>
+	for ForeignChainAliasAccount<AccountId>
+{
+	fn convert_ref(location: impl Borrow<MultiLocation>) -> Result<AccountId, ()> {
+		let entropy = match location.borrow() {
+			// Used on the relay chain for sending paras that use 32 byte accounts
+			MultiLocation {
+				parents: 0,
+				interior: X2(Parachain(para_id), AccountId32 { id, .. }),
+			} => ForeignChainAliasAccount::<AccountId>::from_para_32(para_id, id),
```

Comment:
```suggestion
			} => ForeignChainAliasAccount::<AccountId>::from_relay_32(para_id, id),
```

Reply:
This matching arms matches against messages received on the relay chain coming from a para-chain. I would argue that this should use the conversion from the parachain, allowing to have the same alias on each consuming chain of this converter.

Reply:
Then I would add another item `from_child_32`, and with its own prefix.